### DISCUSSION TITLE
Which .bin file when compiling on Windows

### DISCUSSION THREAD

**Discussion by StefEck23456** [Blackymas/NSPanel_HA_Blueprint#2400] [Q&A]
Hello everyone,

I'm new to HA and I would like to use the Panel with this amazing blueprint.

I tried to compile with ESPHome but it seems that my RP3+ doesn't have sufficient memory. The compilation always stops at main.o. Thats why I tried to compile with Git on my windows computer. The compilation seems to work out well. After flashing the panel and plugging it, the screen remains black when using the firmware.bin. Afterwards I used the firmware.factory.bin, which gives me the original sonoff panel.

I would be happy for any kind of help.

**Comment by edwardtfn** [comment]
Both should work, just the way you install those will define which one you shoukd use... This is how ESPHome is...

Anyways, if you see the original Sonoff screen you are probably in the right direction...
Was the panel detected by Home Assistant as an ESPHome integration? If so, after integrating it to Home Assistant, go to the device's page (Settings > Devices and Services > ESPHome), select the screen model under "Update TFT - Model" and then press the "Update TFT" entity on that same page. It shoukd update the file controlling the screen.

**Reply by edwardtfn** [reply]
You can also adopt it in your ESPHome dashboard (either on your PC or in Home Assistant add-on, as for that the memory requirements are much lower) and try to collect some logs to help troubleshooting.

**Reply by StefEck23456** [reply]
Thanks for your fast response :)

I deleted all compilation files and recompiled. With the sonoff screen visible, I can also find the panel in ESPHome on HA as online. I can also find it in my router connections. 

Unfortunately I'm not able to set it as NSPanel within the blueprint. There is no device to select.

**Reply by edwardtfn** [reply]
Ok, let's try to fix the TFT first, then we look at the Blueprint...
Under your device's page (Settings > Devices & services > ESPHome), could you please select your panels model (EU, US or US Landscape) and then press the "Update TFT" and let me know if the screen changes (after a while transferring the new file)?
Please share the logs in case it fails.

**Reply by StefEck23456** [reply]
The screen changed, thanks!

It also is selectable within the blueprint now. I think I will manage on my own now.

Thanks a lot for your great support.
